modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/DataSetDisplay.js

```diff
@@ -75,29 +78,30 @@ function DataSetDisplay({
 	sorting: sortingProp,
 	style,
 }) {
+	const {apiURL} = useContext(AppContext);
+
 	const wrapperRef = useRef(null);
-	const [dataLoading, setDataLoading] = useState(false);
 	const [componentLoading, setComponentLoading] = useState(false);
-	const [dataSetDisplaySupportSidePanelId] = useState(
-		sidePanelId || 'support-side-panel-' + getRandomId()
-	);
-
+	const [dataLoading, setDataLoading] = useState(!!apiURL);
 	const [dataSetDisplaySupportModalId] = useState(
 		'support-modal-' + getRandomId()
 	);
```

Comment:
Couldn't we use (template literals) instead of concatenation?

```js
const [dataSetDisplaySupportModalId] = useState(
  `support-modal-${getRandomId()}`
);
```